tests/queryset/queryset.py

```diff
@@ -1857,8 +1857,8 @@ class BlogPost(Document):
         self.Person.objects()[:1].delete()
         self.assertEqual(1, BlogPost.objects.count())
 
-    def test_limit_with_write_concern_0(self):
-
+    def test_delete_edge_case_with_write_concern_0_return_None(self):
+        """Return None when write is unacknowledged"""
```

Comment:
Just curious – what's the reason for returning `None` in case of an unack'd write?

Reply:
In the following code:
```
        with set_write_concern(queryset._collection, write_concern) as collection:
            result = collection.delete_many(queryset._query)
            if result.acknowledged:
                return result.deleted_count
```
`result` won't have the `deleted_count` attribute in case the write is unack'd (since its unknown), so I'm just mimicking that.

To make this a bit more explicit, I can change the code to make to the following:
```
        with set_write_concern(queryset._collection, write_concern) as collection:
            result = collection.delete_many(queryset._query)
            if not result.acknowledged:
                # If write concern is set to unacknowledged
                # we dont have the number of deleted items at this point 
                return None
            result.deleted_count
```
let me know what you think

Reply:
> `result` won't have the `deleted_count` attribute in case the write is unack'd (since its unknown)

Ahh, of course, that makes total sense! Thanks for clarifying @bagerard! I added a note about it here: 91899acf.